### PR TITLE
Store signature of SCSnapshot in GlobalSnapshot

### PR DESCRIPTION
**Pull request by marcinwadon**
**IMPORTANT!**
This change is a breaking change for testnet and will require full restart (from genesis) on testnet.
**However, it is not a breaking change for mainnet.**

### CODE REVIEW

File: modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala [hunk 1]
```diff
@@ -315,12 +315,7 @@ object GlobalSnapshotConsensusFunctions {
                 }
                 .getOrElse(Eval.now(List.empty))
 
-            unfold(initLsh).value.toNel.map(nel =>
-              address -> nel.map { event =>
-                StateChannelSnapshotBinary(event.snapshot.value.lastSnapshotHash, event.snapshot.content)
-
-              }.reverse
-            )
+            unfold(initLsh).value.toNel.map(address -> _.map(_.snapshot).reverse)
```
Comment (tbekas) [inline modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala:318]: 👍

**Review by tbekas — APPROVED** [review]
LGTM

**Comment by tbekas** [comment]
Please also adjust the logic of snapshot validation, since now the state channel snapshots can be validated as well.

**Comment by marcinwadon** [comment]
> Please also adjust the logic of snapshot validation, since now the state channel snapshots can be validated as well.

Afaik this is the ticket @kriscieslak is working on right now.

**Pull request merged**